internal/filer_test.go

```diff
@@ -360,100 +361,94 @@ func TestAccFilerReadDir(t *testing.T) {
-	require.NoError(t, err)
-	err = f.Write(ctx, "sqlNb.sql", strings.NewReader("-- Databricks notebook source\n SELECT \"first upload\""))
-	require.NoError(t, err)
-	err = f.Write(ctx, "scalaNb.scala", strings.NewReader("// Databricks notebook source\n println(\"first upload\"))"))
-	require.NoError(t, err)
-	err = f.Write(ctx, "jupyterNb.ipynb", strings.NewReader(jupyterNotebookContent1))
-	require.NoError(t, err)
+	for _, tc := range []struct {
+		name           string
+		nameWithoutExt string
+		content1       string
+		expected1      string
+		content2       string
+	}{
+		{
+			name:           "pyNb.py",
+			nameWithoutExt: "pyNb",
+			content1:       "# Databricks notebook source\nprint('first upload')",
+			expected1:      "# Databricks notebook source\nprint('first upload')",
+			content2:       "# Databricks notebook source\nprint('second upload')",
+		},
+		{
+			name:           "rNb.r",
+			nameWithoutExt: "rNb",
+			content1:       "# Databricks notebook source\nprint('first upload')",
+			expected1:      "# Databricks notebook source\nprint('first upload')",
+			content2:       "# Databricks notebook source\nprint('second upload')",
+		},
+		{
+			name:           "sqlNb.sql",
+			nameWithoutExt: "sqlNb",
+			content1:       "-- Databricks notebook source\n SELECT \"first upload\"",
+			expected1:      "-- Databricks notebook source\n SELECT \"first upload\"",
+			content2:       "-- Databricks notebook source\n SELECT \"second upload\"",
+		},
+		{
+			name:           "scalaNb.scala",
+			nameWithoutExt: "scalaNb",
+			content1:       "// Databricks notebook source\n println(\"first upload\")",
+			expected1:      "// Databricks notebook source\n println(\"first upload\")",
+			content2:       "// Databricks notebook source\n println(\"second upload\")",
+		},
+		{
+			name:           "pythonJupyterNb.ipynb",
+			nameWithoutExt: "pythonJupyterNb",
+			content1:       readFile(t, "testdata/notebooks/py1.ipynb"),
+			expected1:      "# Databricks notebook source\nprint(1)",
+			content2:       readFile(t, "testdata/notebooks/py2.ipynb"),
+		},
+		{
+			name:           "rJupyterNb.ipynb",
+			nameWithoutExt: "rJupyterNb",
+			content1:       readFile(t, "testdata/notebooks/r1.ipynb"),
+			expected1:      "# Databricks notebook source\nprint(1)",
+			content2:       readFile(t, "testdata/notebooks/r2.ipynb"),
+		},
+		{
+			name:           "scalaJupyterNb.ipynb",
+			nameWithoutExt: "scalaJupyterNb",
+			content1:       readFile(t, "testdata/notebooks/scala1.ipynb"),
+			expected1:      "// Databricks notebook source\nprintln(1)",
+			content2:       readFile(t, "testdata/notebooks/scala2.ipynb"),
+		},
+		{
+			name:           "sqlJupyterNotebook.ipynb",
+			nameWithoutExt: "sqlJupyterNotebook",
+			content1:       readFile(t, "testdata/notebooks/sql1.ipynb"),
+			expected1:      "-- Databricks notebook source\nselect 1",
+			content2:       readFile(t, "testdata/notebooks/sql2.ipynb"),
+		},
```

Comment:
I see this is only a refactoring of the test, but why is the returned content for an ipynb notebook not ipynb?

Could you add a comment to the tests why this is the case.

Reply:
Figured it out: the regular workspace files client always uses the `workspace.ExportFormatSource` format.

Reply:
Yeah, added a comment to the test saying as much.